File: Python/day_1.5.py
x = "../inputs/day_1.txt"

with open(x, "r", encoding="utf-8") as f:  # reading the input
    dial: int = 50  # dial starts at 50
    clicks: int = 0
    for line in f:
        line = line.strip()
        direction = line[0]  # 'L' / 'R'
        value = int(line[1:])  # number part
        # print(direction, value) # debug

        if direction == "R":
            dial += value
            clicks += dial // 100
            dial = dial % 100
        else:
            if dial == 0:
                clicks -= 1
            dial -= value
            while dial < 0:
                dial += 100
                clicks += 1
            if dial == 0:
                clicks += 1

        # print(line, "Clicks:", clicks, "Dial:", dial)
    print(clicks)

File: Cpp/day_1.5.cpp
#include <iostream>
#include <fstream>
#include <string>

int main() {
    std::ifstream file("../inputs/day_1.txt");
    std::string line;

    int dial = 50;
    int clicks = 0;

    while (std::getline(file, line)) {
        char direction = line[0];           // 'L' or 'R'
        int value = std::stoi(line.substr(1));  // number part

        if (direction == 'R') {
            dial += value;
            clicks += dial / 100;  // integer division (like // in Python)
            dial = dial % 100;
        } else {  // 'L'
            if (dial == 0) {
                clicks -= 1;
            }
            dial -= value;
            while (dial < 0) {
                dial += 100;
                clicks += 1;
            }
            if (dial == 0) {
                clicks += 1;
            }
        }
    }

    std::cout << clicks << std::endl;
    return 0;
}